tensorflow_addons/rnn/cell.py

```diff
@@ -619,3 +619,176 @@ def get_config(self):
 
         ln_config["layernorm_epsilon"] = ln_config.pop("epsilon")
         return dict(list(cell_config.items()) + list(ln_config.items()))
+
+
+@tf.keras.utils.register_keras_serializable(package="Addons")
+class ESNCell(keras.layers.AbstractRNNCell):
+    """Echo State recurrent Network (ESN) cell.
+
+    This implements the recurrent cell from the paper:
+        H. Jaeger
+        "The "echo state" approach to analysing and training recurrent neural networks". GMD Report148, German National Research Center for Information Technology, 2001.
+
+    Arguments:
+        units: Positive integer, dimensionality in the reservoir.
+        connectivity: Float between 0 and 1.
+            Connection probability between two reservoir units.
+            Default: 0.1.
+        leaky: Float between 0 and 1.
+            Leaking rate of the reservoir.
+            If you pass 1, it's the special case the model does not have leaky integration.
+            Default: 1.
+        spectral_radius: Float between 0 and 1.
+            Desired spectral radius of recurrent weight matrix.
+            Default: 0.9.
+        use_norm2: Boolean, whether to use norm2 as an approximation of the spectral radius.
```

Comment:
The echo state property is "max(eig(W)) < 1". In order to satisfy this property, the weights of the reservoir are scaled based in "spectral_radius/max(eig(W))". However, tensorflow function tf.linalg.eig has an exponential complexity, so I decided to add an alternative implementation using an approximation.
norm2 refers to the [p-norm](https://en.wikipedia.org/wiki/Matrix_norm#Special_cases) function with p=2. This can be used as an upper bound of the spectral radius so that the echo state property is satisfied. Check [Wikipedia/MaxNorm](https://en.wikipedia.org/wiki/Matrix_norm) and this [paper] where they use the 2-norm for this same purpose.

What do you think should be added to the docstring  about this?

Reply:
Ack, I think it wasn't clear to me that "norm2" is p-norm where p=2. So probably it need some clearification.

Reply:
Improved the description, please check it.